src/amber/server/env.cr

```diff
@@ -1,33 +1,38 @@
 module Amber
   def self.env
-    Env.new(Settings.env.not_nil!.downcase)
+    @@env ||= Env.new(ENV["AMBER_ENV"]? || "development") # Would be best to only define this once.
   end
 
   class Env
-    def initialize(@env : String | Symbol)
+    alias EnvType = String | Symbol 
+
+    def initialize(@env : String) end
+
+    def is?(environment : EnvType)
+      @env == environment.to_s.downcase
     end
 
-    def in?(environment_list : Array(String | Symbol))
-      (environment_list.map &.to_s.downcase).includes? @env
+    def in?(environment_list : Array(EnvType))
+      environment_list.each do |other_env|
```

Comment:
Could have just removed the map &.to_s.downcase since includes?, any? Does the same thing as above

Reply:
Also each check is down casing the other_env which is more operations than downcasing everything at once. Again don’t understand the performance gain here

Reply:
In the past `&.map.to_s.downcase` was also modifying the strings one at a time. In this case we're just relying on is? or == to do the to_s.downcasing which is DRYer. Also in the case that the first one matches it doesn't go on to the second or third and no string operations are performed.

Reply:
What do you think of 

```crystal
environment_list.any? { |env| self.== env.to_s.downcase }
```

Reply:
Does any? return as soon as it finds one?

Reply:
Yes it exits when the first instance  is found. See https://github.com/crystal-lang/crystal/blob/e2a1389e8165fb097c785524337d7bb7b550a086/src/enumerable.cr#L68

Reply:
Perfect. Thanks.